### ISSUE TITLE
Build failure on production due to concurrent-ruby gem

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#3086]
I just tried publishing to production and saw this error:

```
Installing will_paginate 3.1.6
RuntimeError: Digest::Base cannot be directly inherited in Ruby
An error occurred while installing concurrent-ruby (1.0.5), and Bundler cannot
continue.
Make sure that `gem install concurrent-ruby -v '1.0.5'` succeeds before
bundling.

In Gemfile:
  geokit-rails was resolved to 2.3.1, which depends on
    rails was resolved to 5.2.0, which depends on
      actioncable was resolved to 5.2.0, which depends on
        actionpack was resolved to 5.2.0, which depends on
          actionview was resolved to 5.2.0, which depends on
            rails-dom-testing was resolved to 2.0.3, which depends on
              activesupport was resolved to 5.2.0, which depends on
                i18n was resolved to 1.0.1, which depends on
                  concurrent-ruby
ERROR: Service 'web' failed to build: The command '/bin/sh -c bundle install --jobs 4' returned a non-zero code: 5
Makefile:8: recipe for target 'redeploy-container' failed
make: *** [redeploy-container] Error 1

```

@icarito would you know what's going on here? Production didn't finish building, so it's OK, just at the last version.

**Comment by icarito** [comment]
https://github.com/chef/chef/issues/1793 suggests this is an issue with Digest gem not being thread safe...? Researching...

**Comment by icarito** [comment]
It is suggested to use `OpenSSL::Digest instead`.

**Comment by jywarren** [comment]
Here's an example of a PR merging this change: https://github.com/chef/chef/commit/d5f16cf0b9f35cbdff52ab83c221666e6843287e

**Comment by icarito** [comment]
This gave issues in production but not in Staging? How could that be the case? :-/

**Comment by jywarren** [comment]
Maybe needed on this line? 

https://github.com/publiclab/plots2/blob/3254f3b79167a36fd367c37f288e733330829bbf/app/models/user.rb#L356

**Comment by jywarren** [comment]
weirdness. i opened a PR...

**Comment by icarito** [comment]
I also found a mention here:

https://github.com/publiclab/plots2/blob/ed1ef3b72c8a45bfd1d7b842d1d747db391b2a7f/doc/app/User.html#L1083

Still it doesn't explain why it would fail in production and not staging.

**Comment by icarito** [comment]
the only difference I can think of is the forking configuration: 

https://github.com/publiclab/plots2/blob/master/containers/docker-compose-production.yml#L22

Perhaps this triggers it? But it failed during Gem install process, not on application startup, so the startup command shouldn't matter.

**Comment by icarito** [comment]
Let me know if #3088 fixes the issue otherwise https://github.com/publiclab/plots2/blob/ed1ef3b72c8a45bfd1d7b842d1d747db391b2a7f/doc/app/User.html#L1083 could also be needed. I'm not experienced dealing with Ruby internals. Wierd that it didn't happen in staging!

**Comment by jywarren** [comment]
Solved in #3088 !
